ccm_web/client/src/utils/enrollmentValidators.ts

```diff
@@ -0,0 +1,113 @@
+import { ClientEnrollmentType, isValidRole } from '../models/canvas'
+import { InvalidationType } from '../models/models'
+import { DuplicateIdentifierInRowsValidator, RowInvalidation } from '../utils/rowValidation'
+import {
+  emailSchema, firstNameSchema, lastNameSchema, loginIDSchema, validateString, ValidationResult
+} from '../utils/validation'
+
+export interface EnrollmentInvalidation extends RowInvalidation {}
+
+interface EnrollmentRowsValidator {
+  validate: (values: string[]) => EnrollmentInvalidation[]
+}
+
+const getMessage = (result: ValidationResult, fieldName: string): string => {
+  return result.messages.length > 0 ? result.messages[0] : `Value for ${fieldName} is invalid.`
+}
+
+export class EmailRowsValidator implements EnrollmentRowsValidator {
+  validate (emails: string[]): EnrollmentInvalidation[] {
+    const invalidations: EnrollmentInvalidation[] = []
+    emails.forEach((email, i) => {
+      const emailValidationResult = validateString(email, emailSchema)
+      if (!emailValidationResult.isValid) {
+        invalidations.push({
+          rowNumber: i + 2,
+          message: getMessage(emailValidationResult, 'email address'),
+          type: InvalidationType.Error
+        })
+      }
+    })
+    return invalidations
+  }
+}
+
+export class DuplicateEmailRowsValidator implements EnrollmentRowsValidator {
+  validate (emails: string[]): EnrollmentInvalidation[] {
+    const dupEmailValidator = new DuplicateIdentifierInRowsValidator('email address')
```

Comment:
can you make the `email address` as  const here since the string is used in more that 2 places

Reply:
Optional: same goes for `first name, last name, loginID` but since only one file that is fine leave off

Reply:
Can I postpone doing this, @pushyamig? It may make sense to put these constants in a different file, since I'm already using them in `validation`. Or maybe it makes sense to pair the `yup` schemas with the string, like this: 

```
interface FieldSchemaData {
  fieldName: string,
  schema: StringSchema
}
```

Anyways, this doesn't feel like it should be a blocker to merging this, right?

Reply:
Sure.